### PR TITLE
fix: Claims Retrieval Issue in SqlIdentityHubStore Due to Missing pay…

### LINKED ISSUE #107: Claims Retrieval Issue in SqlIdentityHubStore Due to Missing payloadFormat
# Bug Report

## Describe the Bug
payloadFormat is not being added in the SqlIdentityHubStore and therefore the claims cannot be retrieved in the other functions.

### Expected Behavior
When the sql store of identity hub is being used, the claims should be retrieved.

### Observed Behavior
When the sql store of identity hub is being used, the claims cannot be retrieved.

## Detailed Description
The issue with the current implementation of the SqlIdentityHubStore is that the payloadFormat is not being added, resulting in the inability to retrieve claims in other functions.
To resolve this issue, the payloadFormat needs to be included in the code of the SqlIdentityHubStore. One possible implementation for this is to add the line of code from: Possible Implementation
It is important to thoroughly test this solution to ensure that it resolves the problem and does not introduce any unforeseen issues.
Please note that this detailed description provides an overview of the issue and suggests a potential solution. Further investigation and analysis may be required to fully address the problem.

## Possible Implementation
.payloadFormat(resultSet.getString(statements.getPayloadFormatColumn()))

### PR DESCRIPTION
**Pull request by konrad1adamczyk**
fix: Claims Retrieval Issue in SqlIdentityHubStore Due to Missing payloadFormat #107

Closes #107

### CODE REVIEW

File: extensions/store/sql/identity-hub-store-sql/src/main/java/org/eclipse/edc/identityhub/store/sql/SqlIdentityHubStore.java [hunk 1]
```diff
@@ -71,6 +71,7 @@ public void add(IdentityHubRecord record) {
     private IdentityHubRecord parse(ResultSet resultSet) throws SQLException {
         return IdentityHubRecord.Builder.newInstance()
                 .id(resultSet.getString(statements.getIdColumn()))
+                .payloadFormat(resultSet.getString(statements.getPayloadFormatColumn()))
```
Comment (paullatzelsperger) [inline extensions/store/sql/identity-hub-store-sql/src/main/java/org/eclipse/edc/identityhub/store/sql/SqlIdentityHubStore.java:74]: could you add a test for this, to guard against regression?

**Review by paullatzelsperger — CHANGES_REQUESTED** [review]

**Comment by paullatzelsperger** [comment]
thanks for the fix, but please familiarize yourself with our [pr etiquette](https://github.com/eclipse-edc/Connector/blob/main/pr_etiquette.md), which states that raising PRs without associated issues is not allowed. After having raised an issue, it should go through triage before any work can happen.

**Comment by bscholtes1A** [comment]
LGTM, I'll approve when unit-test is added.

**Comment by paullatzelsperger** [comment]
@konrad1adamczyk postgres tests are failing. can you see what's going on there?

**Comment by konrad1adamczyk** [comment]
yes I see it here, on my local machine it was passing, I will fix

**Comment by paullatzelsperger** [comment]
checkstyle is failing due to a missing license header.

File: extensions/store/sql/identity-hub-store-sql/src/test/java/org/eclipse/edc/identityhub/store/sql/SqlIdentityHubStoreTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+
+    @Test
+    public void testParse() throws SQLException {
+        when(resultSet.getString("id")).thenReturn("123");
+        when(resultSet.getString("payloadFormat")).thenReturn("json");
+        when(resultSet.getString("payload")).thenReturn("{\"key\": \"value\"}");
+        when(resultSet.getLong("created_at")).thenReturn(1625385600000L);
+
+        IdentityHubRecord result = sqlIdentityHubStore.parse(resultSet);
```
Comment (paullatzelsperger) [inline extensions/store/sql/identity-hub-store-sql/src/test/java/org/eclipse/edc/identityhub/store/sql/SqlIdentityHubStoreTest.java:47]: user `var`

File: extensions/store/sql/identity-hub-store-sql/src/test/java/org/eclipse/edc/identityhub/store/sql/SqlIdentityHubStoreTest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+        assertEquals("123", result.getId());
```
Comment (paullatzelsperger) [inline extensions/store/sql/identity-hub-store-sql/src/test/java/org/eclipse/edc/identityhub/store/sql/SqlIdentityHubStoreTest.java:49]: please use `assertJ` assertions

File: extensions/store/sql/identity-hub-store-sql/src/main/java/org/eclipse/edc/identityhub/store/sql/SqlIdentityHubStore.java [hunk 4]
```diff
@@ -68,9 +68,10 @@ public void add(IdentityHubRecord record) {
         });
     }
 
-    private IdentityHubRecord parse(ResultSet resultSet) throws SQLException {
+    IdentityHubRecord parse(ResultSet resultSet) throws SQLException {
```
Comment (paullatzelsperger) [inline extensions/store/sql/identity-hub-store-sql/src/main/java/org/eclipse/edc/identityhub/store/sql/SqlIdentityHubStore.java:71]: this method is `private`. making methods `package-private` or `protected` simply to be able to test them is an anti-pattern.

File: extensions/store/sql/identity-hub-store-sql/src/test/java/org/eclipse/edc/identityhub/store/sql/SqlIdentityHubStoreTest.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import org.mockito.MockitoAnnotations;
+
+import java.io.IOException;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+public class SqlIdentityHubStoreTest {
```
Comment (paullatzelsperger) [inline extensions/store/sql/identity-hub-store-sql/src/test/java/org/eclipse/edc/identityhub/store/sql/SqlIdentityHubStoreTest.java:20]: this is not a sufficient test for the `SqlIdentityHubStore`.

File: extensions/store/sql/identity-hub-store-sql/src/test/java/org/eclipse/edc/identityhub/store/sql/SqlIdentityHubStoreTest.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    private SqlIdentityHubStore sqlIdentityHubStore;
+
+    @BeforeEach
+    void setUp() throws IOException {
+        var statements = new BaseSqlIdentityHubStatements();
+        var typeManager = new TypeManager();
+        var dataSourceRegistry = new DefaultDataSourceRegistry();
+        var transactionContext = new NoopTransactionContext();
+
+        MockitoAnnotations.openMocks(this);
```
Comment (paullatzelsperger) [inline extensions/store/sql/identity-hub-store-sql/src/test/java/org/eclipse/edc/identityhub/store/sql/SqlIdentityHubStoreTest.java:34]: not how we do mocks. please use the static `mock()` method. also, please use static imports.

**Review by paullatzelsperger — CHANGES_REQUESTED** [review]
Please familiarize yourself with our [coding principles](https://github.com/eclipse-edc/Connector/blob/main/docs/developer/architecture/coding-principles.md) and [testing rules](https://github.com/eclipse-edc/Connector/blob/main/docs/developer/testing.md). 

Maybe you could add unit and integration tests (using Testcontainers) for the SqlIdentityHubStore, it would a) be a good way to familiarize yourself with the codebase and our principles, and b) would be more comprehensive test. Unfortunately we cannot accept this PR as-is.

**Comment by bscholtes1A** [comment]
handled through #112

**Pull request closed without merging**